Modules/posixmodule.c

```diff
@@ -12020,13 +12020,22 @@ typedef struct {
 #endif
 } DirEntry;
 
+static PyObject *
+DirEntry_new(PyTypeObject *type, PyObject *args, PyObject *kwargs)
```

Comment:
Would not enough to set tp_new to NULL after calling PyType_FromSpec() as it was done in other places (see curses_panel and _tkinter modules for examples).

Reply:
Yes that should be enough. However, in order to comply with PEP384, PyTypeObjects should be opaque. Providing this tp_new results in the exact same behavior as setting it to NULL with the added benefit that we don't access the slot directly! :)